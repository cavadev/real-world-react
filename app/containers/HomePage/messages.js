/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.HomePage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the HomePage container!',
  },
  boilerplate: {
    id: `${scope}.boilerplate`,
    defaultMessage:
      'Example of real world project with ReactJS. The first commit of this project was built with ',
  },
  realWorld: {
    id: `${scope}.realWorld`,
    defaultMessage: 'You can see more of the project in ',
  },
  account: {
    id: `${scope}.account`,
    defaultMessage: 'The next page is only visible for logged users: ',
  },
});
