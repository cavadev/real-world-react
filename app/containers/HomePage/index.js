/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

import StyledCard from 'components/css/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

import messages from './messages';

export default function HomePage() {
  return (
    <div>
      <StyledCard>
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            <FormattedMessage {...messages.header} />
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            <FormattedMessage {...messages.boilerplate} />
            <a
              href="https://github.com/react-boilerplate/react-boilerplate"
              target="_blank"
            >
              [react-boilerplate]
            </a>
            .
            <br />
            <FormattedMessage {...messages.account} />
            <Link to="/account"> account</Link>
          </Typography>
        </CardContent>

        <CardActions>
          <Button
            size="small"
            color="primary"
            href="https://github.com/cavadev/real-world-react"
          >
            Github
          </Button>
        </CardActions>
      </StyledCard>
    </div>
  );
}
