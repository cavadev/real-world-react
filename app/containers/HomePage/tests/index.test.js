import React from 'react';
// BrowserRouter is to avoid error: not use <Link> outside a <Router>
import { BrowserRouter } from 'react-router-dom';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';

import HomePage from '../index';

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
