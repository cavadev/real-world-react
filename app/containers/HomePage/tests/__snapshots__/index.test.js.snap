// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiCard-root Card__StyledCard-sc-1kvgq7c-0 iCqqgf MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        <span>
          This is the HomePage container!
        </span>
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        <span>
          Example of real world project with ReactJS. The first commit of this project was built with 
        </span>
        <a
          href="https://github.com/react-boilerplate/react-boilerplate"
          target="_blank"
        >
          [react-boilerplate]
        </a>
        .
        <br />
        <span>
          The next page is only visible for logged users: 
        </span>
        <a
          href="/account"
        >
           account
        </a>
      </p>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing"
    >
      <a
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
        href="https://github.com/cavadev/real-world-react"
        role="button"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Github
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</div>
`;
